include/nbl/system/IFile.h

```diff
@@ -37,6 +37,10 @@ class IFile : public core::IReferenceCounted
 	virtual void* getMappedPointer() = 0;
 	virtual const void* getMappedPointer() const = 0;
 
+	virtual void seek(uint32_t pos) = 0;
+	virtual void read(char* outData, size_t count) const;
+	virtual size_t getPos() const = 0;
```

Comment:
nah, we want to get rid of this kind of API. User must use ISystem API (`readFile`, `writeFile`) to access file contents

Reply:
You can have pure virtual methods for file content access here (i'd say it's encouraged implementation approach since we want to have memory files as well), but they should not be in public IFile API + avoid `seek()`-like approaches